tests/test_TFUtil.py

```diff
@@ -469,6 +469,30 @@ def test_Data_copy_compatible_to_batch_feature_is_dynamic():
   assert_equal(t.get_size_dim_tag(0), dec.get_time_dim_tag())
 
 
+def test_Data_get_common_data_modified_time_dim():
+  d1 = Data(name='t', shape=(None, 32, 128), dtype='float32', auto_create_placeholders=True)
+  d2 = Data(name='r', shape=(None, 32, 128), dtype='float32', auto_create_placeholders=True)
+  # simulating the changes in the time dim
+  d1.size_placeholder[0] -= 1
+  d2.size_placeholder[0] -= 1
+
+  from TFUtil import DimensionTag
+  # doing tagging a la SliceLayer for d1.size_placeholder
+  if not DimensionTag.get_tag_from_size_tensor(d1.size_placeholder[0]):
```

Comment:
Why this check? You are constructing the test case. And before the `size_placeholder[0] -= 1`, it already has a valid dim tag, so you don't need to set a new one. Just remove the `size_placeholder[0] -= 1`. And then also this whole `if` branch.